apollo-router/src/context/extensions/sync.rs

```diff
@@ -24,14 +24,29 @@ impl ExtensionsMutex {
     /// It is CRITICAL to avoid holding on to the mutex guard for too long, particularly across async calls.
     /// Doing so may cause performance degradation or even deadlocks.
     ///
+    /// DEPRECATED: prefer with_lock()
+    ///
     /// See related clippy lint for examples: <https://rust-lang.github.io/rust-clippy/master/index.html#/await_holding_lock>
+    #[deprecated]
```

Comment:
Q: Is there a reason we don't use the Deprecated macro's parameters on things like this here?  Doesn't that provide a better user experience?:

```
#[deprecated(since = "1.49.0", note = "Users should instead use with_lock")]
```

Reply:
It definitely renders nicer in the documentation!